Modules/Devices/test/Devices.Application.Tests/Tests/Identities/Queries/GetDeletionProcessesAuditLogs/FindDeletionProcessAuditLogsByAddressStubRepository.cs

```diff
@@ -18,42 +18,42 @@ public FindDeletionProcessAuditLogsByAddressStubRepository(IEnumerable<IdentityD
 
     public Task<bool> Exists(IdentityAddress address, CancellationToken cancellationToken)
     {
-        throw new NotImplementedException();
+        throw new NotSupportedException();
```

Comment:
> By switching to this, we'll be adding more rules

What rules do you mean?

> Why would we do this? The default body when methods are created by the IDE is `throw new NotImplementedException();`

It is, yes. And that's why there were so many occurrences of it (=> I kept it that way, too). But I recently used Rider to find all open todos in my code, and the window showed me not only the TODO comments, but also every place where a `NotImplementedException` was thrown. That's when I realized that a `NotSupportedException` might be a better choice. Because a `NotImplementedException` should probably only be a temporary solution. When you find a place in the code that throws such an exception, you shouldn't have to ask yourself "Is this on purpose or did someone forget to implement it?"

Reply:
I'm referring to the unwritten rule of what Exception to throw in Stub Repositories.

As for the rest, I agree that makes sense. However, a `NotSupportedException` may also be deceiving as many of those could actually be implemented, i.e., they're not like that by design. This raises the question: why don't we use FakeItEasy to setup mock/stub repos instead of having a huge file with only one implemented method?

Reply:
> However, a NotSupportedException may also be deceiving as many of those could actually be implemented, i.e., they're not like that by design.

Yes, but I do actually think it makes sense, because **the current implementation** does not support it. 🤔 But let's discuss that in tomorrows daily. I'm curious what the others think about it.

> This raises the question: why don't we use FakeItEasy to setup mock/stub repos instead of having a huge file with only one implemented method?

There are some cases where "custom" stubs are easier to set up and understand than ones created by FakeItEasy.